scikeras/utils/__init__.py

```diff
@@ -0,0 +1,109 @@
+from inspect import isclass
+from typing import Callable, Union
+
+import numpy as np
+
+from tensorflow.keras.losses import Loss
+from tensorflow.keras.losses import get as get_loss
+from tensorflow.keras.losses import serialize as serialize_loss
+from tensorflow.keras.metrics import Metric
+from tensorflow.keras.metrics import get as get_metric
+from tensorflow.keras.metrics import serialize as serialize_metric
+
+
+def loss_name(loss: Union[str, Loss, Callable]) -> str:
+    """Retrieves a loss's full name (eg: "mean_squared_error").
+
+    Parameters
+    ----------
+    loss : Union[str, Loss, Callable]
+        Instance of Keras Loss, loss callable or string
+        shorthand (eg: "mse") or full name ("mean_squared_error").
+
+    Returns
+    -------
+    str
+        [description]
+
+    Examples
+    --------
+    >>> loss_name("BinaryCrossentropy")
+    'BinaryCrossentropy'
+    >>> loss_name("binary_crossentropy")
+    'binary_crossentropy'
+    >>> import tensorflow.keras.losses as losses
+    >>> loss_name(losses.BinaryCrossentropy)
+    'BinaryCrossentropy'
+    >>> loss_name(losses.binary_crossentropy)
+    'binary_crossentropy'
+
+    Raises
+    ------
+    ValueError
+        In case of an unknown loss.
+    """
+    if isclass(loss):
+        try:
+            loss = loss()  # get_loss accepts instances, not classes
+        except TypeError:
+            # convert TypeError -> ValueError for consistency with errors
+            # raised by get_loss
+            raise ValueError(f"Unknown loss: {loss}")
```

Comment:
The loss isn't unknown; it can't be called with `loss()`. Maybe make the error message "pass an instantiated loss instead"?

Reply:
I think the loss is unknown in the sense that `we don't know how to process this loss`. I would like to keep the results of this method abstract from its implementation, and it should fail with a singular error type and message, that way it's easy to call and catch. Maybe we should have `ValueError: Unable to process loss: {loss}`?

Reply:
>  it should fail with a singular error type and message, 

:-1: yes, that will make testing easier. I don't think that'll make it any easier for users:

``` python
try:
    loss = loss_name("mse")
except:
    loss = None
```

Reply:
Let's leave the discussion about loss/metric name aside for now. I'll suggest a new implementation in a PR.

Reply:
Happy to leave it for another PR, but I think we're almost done. I resolved/will resolve all of the issues in https://github.com/adriangb/scikeras/pull/88#discussion_r502843705, and I think this resolves this thread as well, right?

Reply:
I think those resolve most of them. I'd also add a TypeError at the top of `loss_name` for lists/dicts given https://github.com/adriangb/scikeras/pull/88#discussion_r502830348.

Reply:
I also think it'd be good to have a bidirectional map between loss shorthands and classes. I don't think a user would care if "acc" gets renamed to "accuracy."

Reply:
That I think we can certainly leave for another PR.

Reply:
I am not a fan of catching errors and raising "one type of error with one message." I believe that will confuse users.

Reply:
> I am not a fan of catching errors and raising "one type of error with one message."

Are you referring to the type of error (which I think is what this thread was about, and is largely resolved) or the error message (which we are discussing in https://github.com/adriangb/scikeras/pull/88#discussion_r502819477)?

Reply:
Resolved in #102